### PR TITLE
[REEF-932] ConfigurationModule.build() should log missing parameters on failure

### PR DESCRIPTION
**Pull request by zerg-junior**
This patch:
 * Adds the descriptive message to the IllegalStateException thrown by ConfigurationModule.build()
 * Adds a logger to ConfigurationModule and logs the exception
 * Adds a test to ensure that the exception or its message are not deleted by accident in future versions

JIRA:
 [REEF-932] (https://issues.apache.org/jira/browse/REEF-932)

### CODE REVIEW

File: lang/java/reef-tang/tang/src/main/java/org/apache/reef/tang/formats/ConfigurationModule.java [hunk 1]
```diff
@@ -47,6 +49,10 @@
  * See org.apache.reef.tang.formats.TestConfigurationModule for more information and examples.
  */
 public class ConfigurationModule {
+
+  private static final String CLASS_NAME = ConfigurationModule.class.getName();
```
Comment (markusweimer) [inline lang/java/reef-tang/tang/src/main/java/org/apache/reef/tang/formats/ConfigurationModule.java]: Why the added field for the `CLASS_NAME`? Can't we just inline this call in the `getLogger()` call?

**Comment by markusweimer** [comment]
This looks good (besides a minor comment). However, it would be good to have a test for this that makes sure we never loose this functionality by accident.

**Comment by zerg-junior** [comment]
@markusweimer I'll move the 
```
   for (final Class<? extends Name<?>> clazz : c.builder.freeParams.keySet()) 
```
loop to a separate method and test the method. This will help readability as well.

**Comment by zerg-junior** [comment]
@markusweimer Updated

**Comment by markusweimer** [comment]
@zerg-junior I'll test and merge.

**Pull request closed without merging**